core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1486,9 +1486,28 @@ public void delete(AlluxioURI path, DeleteOptions options) throws IOException,
         LockedInodePath inodePath =
             mInodeTree.lockInodePath(lockingScheme.getPath(), lockingScheme.getMode());
         FileSystemMasterAuditContext auditContext =
-            createAuditContext("delete", path, null, inodePath.getInodeOrNull())) {
+            createAuditContext("delete", path, null, inodePath.getInodeOrNull());
+        LockedInodePathList children =
+            options.isRecursive() ? mInodeTree.lockDescendants(inodePath, InodeTree.LockMode.WRITE)
+                : null) {
       try {
         mPermissionChecker.checkParentPermission(Mode.Bits.WRITE, inodePath);
```

Comment:
should we also check permissions for the path being deleted? It looks like we check its parent and its children, but we never check the path itself

Reply:
In posix, you only need write permissions on the parent, not the path itself. I'm not sure what hdfs does.

Reply:
oh interesting, didn't realize it worked like that

Reply:
verified w/ hdfs as well. you can delete if you parent access but not access to the path being deleted